clients/venice-admin-tool/src/main/java/com/linkedin/venice/AdminTool.java

```diff
@@ -1803,14 +1804,33 @@ private static void deleteKafkaTopic(CommandLine cmd, PubSubClientsFactory pubSu
 
   private static void dumpAdminMessages(CommandLine cmd, PubSubClientsFactory pubSubClientsFactory) {
     Properties consumerProperties = loadProperties(cmd, Arg.KAFKA_CONSUMER_CONFIG_FILE);
+    VeniceProperties veniceProperties = new VeniceProperties(consumerProperties);
     String pubSubBrokerUrl = getRequiredArgument(cmd, Arg.KAFKA_BOOTSTRAP_SERVERS);
     consumerProperties = DumpAdminMessages.getPubSubConsumerProperties(pubSubBrokerUrl, consumerProperties);
+    PubSubPositionTypeRegistry pubSubPositionTypeRegistry =
+        PubSubPositionTypeRegistry.fromPropertiesOrDefault(veniceProperties);
+    PubSubPositionDeserializer pubSubPositionDeserializer = new PubSubPositionDeserializer(pubSubPositionTypeRegistry);
     PubSubConsumerAdapter consumer = getConsumer(consumerProperties, pubSubClientsFactory);
+    String startingOffset = getOptionalArgument(cmd, Arg.STARTING_OFFSET);
```

Comment:
We should Apache Commons CLI’s OptionGroup for mutually exclusive args (e.g., --starting-offset vs --starting-position) instead of manual validation.

Benefits:
	•	Built-in mutual exclusion handling
	•	Cleaner error messages
	•	Less boilerplate validation code
	•	More consistent and less error-prone

This pattern will show up in multiple places in AdminTool.java, so switching to OptionGroup could simplify and standardize the CLI code.

Reply:
TBH we should just show offset related command arguments as deprecated instead and gradually remove it.
It should only exist during the migration phase and should be phased out. Regarding the usage of OptionGroup, that could be a general plus, but can happen in a separate PR

Reply:
Right. We can achieve this without duplicating logic at multiple places.